txnsync/transactionCache.go

```diff
@@ -68,13 +77,33 @@ type longTermTransactionCache struct {
 	timestamps      []time.Duration
 }
 
+// detach remove the entry from the list it's currently part of.
+// the return value is false if the item is the only entry in the list
+// or true otherwise.
+func (ce *shortTermCacheEntry) detach() bool {
+	if ce.next == ce.prev {
+		return false
+	}
+	ce.prev.next = ce.next
+	ce.next.prev = ce.prev
+	return true
+}
+
+// addToList add the element to the tail of the list who's head is firstListEntry.
+func (ce *shortTermCacheEntry) addToList(firstListEntry *shortTermCacheEntry) {
```

Comment:
firstListEntry -> head, as the most common name, so it would eliminate half of the comment :)